core/monitor/monitor.go

```diff
@@ -19,32 +19,50 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/golang/glog"
-
+	"github.com/google/keytransparency/core/client/mutationclient"
 	"github.com/google/keytransparency/core/monitorstorage"
-	ktpb "github.com/google/keytransparency/core/proto/keytransparency_v1_proto"
 
 	"github.com/google/trillian"
 	"github.com/google/trillian/client"
-	tcrypto "github.com/google/trillian/crypto"
 	"github.com/google/trillian/crypto/keys/der"
 	"github.com/google/trillian/merkle/hashers"
+
+	"github.com/golang/glog"
+
+	gpb "github.com/google/keytransparency/core/proto/keytransparency_v1_grpc"
+	pb "github.com/google/keytransparency/core/proto/keytransparency_v1_proto"
+	tcrypto "github.com/google/trillian/crypto"
 )
 
 // Monitor holds the internal state for a monitor accessing the mutations API
 // and for verifying its responses.
 type Monitor struct {
+	mclient gpb.MutationServiceClient
+	signer  *tcrypto.Signer
+	trusted *trillian.SignedLogRoot
+
 	mapID       int64
-	mapHasher   hashers.MapHasher
-	mapPubKey   crypto.PublicKey
 	logVerifier client.LogVerifier
-	signer      *tcrypto.Signer
-	trusted     *trillian.SignedLogRoot
 	store       monitorstorage.Interface
+	mapHasher   hashers.MapHasher
+	mapPubKey   crypto.PublicKey
 }
 
-// New creates a new instance of the monitor.
-func New(logverifierCli client.LogVerifier, mapTree *trillian.Tree, signer *tcrypto.Signer, store monitorstorage.Interface) (*Monitor, error) {
+// NewFromConfig produces a new monitor from a DomainInfo object.
+func NewFromConfig(mclient gpb.MutationServiceClient,
```

Comment:
I'd stick with the single line declaration.  Line length doesn't appear to be very excessive.
The layout you have indents the argument continuation lines the same as the code block following which is harder to read.